File: ptm_task/src/core/domain/dto/response/note_response.go
/*
Author: QuanTuanHuy
Description: Part of Serp Project
*/

package response

import "github.com/serp/ptm-task/src/core/domain/entity"

type NoteResponse struct {
	ID       int64 `json:"id"`
	UserID   int64 `json:"userId"`
	TenantID int64 `json:"tenantId"`

	TaskID    *int64 `json:"taskId,omitempty"`
	ProjectID *int64 `json:"projectId,omitempty"`

	Content      string  `json:"content"`
	ContentPlain *string `json:"contentPlain,omitempty"`

	Attachments []entity.NoteAttachment `json:"attachments,omitempty"`

	IsPinned     bool   `json:"isPinned"`
	ActiveStatus string `json:"activeStatus"`

	CreatedAt int64 `json:"createdAt"`
	UpdatedAt int64 `json:"updatedAt"`

	TaskTitle    *string `json:"taskTitle,omitempty"`
	ProjectTitle *string `json:"projectTitle,omitempty"`
}

type NoteListResponse struct {
	Notes      []*NoteResponse `json:"notes"`
	TotalCount int64           `json:"totalCount"`
	Limit      int             `json:"limit"`
	Offset     int             `json:"offset"`
}

type NoteSearchResponse struct {
	Notes      []*NoteResponse `json:"notes"`
	TotalCount int64           `json:"totalCount"`
	Query      string          `json:"query"`
}

File: crm/src/main/java/serp/project/crm/core/domain/dto/response/NoteResponse.java
/**
 * Author: QuanTuanHuy
 * Description: Part of Serp Project
 */

package serp.project.crm.core.domain.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class NoteResponse {
    private Long id;
    private Long tenantId;
    private String entityType;
    private Long entityId;
    private String content;
    private Long createdAt;
    private Long updatedAt;
    private Long createdBy;
    private Long updatedBy;
}
